caikit_nlp/modules/token_classification/transformer_sentence_classification.py

```diff
@@ -0,0 +1,218 @@
+            threshold: float
+                (Optional) Threshold based on which to return score results
+
+        Returns:
+            TokenClassificationResult
+        """
+        if threshold is None:
+            threshold = self.default_threshold
+        token_classification_results = []
+        # Split document into sentences
+        sentence_span_list = self.sentence_splitter.run(text)
+        # Run each sentence through the classifier and determine based
+        # on threshold and labels_to_output what results should be returned
+        text_list = [span.text for span in sentence_span_list]
+        classification_results = self.sequence_classifier.run_batch(text_list)
+        for idx, classification_result in enumerate(classification_results):
+            # Each classification prediction is list of classifications
+            # for that particular text example
+            for classification in classification_result.results:
+                # Map labels to semantic labels if provided
+
+                # NOTE: labels need to be specified as str for config
+                label = str(classification.label)
+                if self.labels_mapping:
+                    # Use original classifier label if not found
+                    label = self.labels_mapping.get(label, label)
+                if classification.score >= threshold:
+                    if not self.labels_to_output or (
+                        self.labels_to_output and label in self.labels_to_output
+                    ):
+                        sentence_span = sentence_span_list[idx]
+                        token_classification = TokenClassification(
+                            start=sentence_span.start,
+                            end=sentence_span.end,
+                            word=sentence_span.text,
+                            entity=label,
+                            score=classification.score,
+                        )
+                        token_classification_results.append(token_classification)
+        return TokenClassificationResult(results=token_classification_results)
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        module_saver = ModuleSaver(
+            self,
+            model_path=model_path,
+        )
+
+        with module_saver:
+            module_saver.save_module(self.sentence_splitter, "sentence_split")
+            module_saver.save_module(
+                self.sequence_classifier, "sequence_classification"
+            )
+            config_options = {
+                "language": self.lang,
+                "default_threshold": self.default_threshold,
+                "labels_mapping": self.labels_mapping,
+                "labels_to_output": self.labels_to_output,
+            }
+            module_saver.update_config(config_options)
```

Comment:
This will never get called right? Since we don't have bootstrap or train functions?

Reply:
Hm it is definitely still possible to initialize (via init or load) and save this module. We have used this in order to create artifacts with different underlying sentence splitter and sequence classification modules.

Reply:
Discussed offline - resolution here is to add `bootstrap` as the entrypoint for both modules